crates/router/src/connector/powertranz/transformers.rs

```diff
@@ -1,210 +1,336 @@
 
-//TODO: Fill the struct with respective fields
 // Auth Struct
 pub struct PowertranzAuthType {
-    pub(super) api_key: Secret<String>,
+    pub(super) power_tranz_id: Secret<String>,
+    pub(super) power_tranz_password: Secret<String>,
 }
 
 impl TryFrom<&types::ConnectorAuthType> for PowertranzAuthType {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
         match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: Secret::new(api_key.to_string()),
+            types::ConnectorAuthType::BodyKey { api_key, key1 } => Ok(Self {
+                power_tranz_id: Secret::new(key1.to_string()),
+                power_tranz_password: Secret::new(api_key.to_string()),
             }),
             _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum PowertranzPaymentStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
-}
-
-impl From<PowertranzPaymentStatus> for enums::AttemptStatus {
-    fn from(item: PowertranzPaymentStatus) -> Self {
-        match item {
-            PowertranzPaymentStatus::Succeeded => Self::Charged,
-            PowertranzPaymentStatus::Failed => Self::Failure,
-            PowertranzPaymentStatus::Processing => Self::Authorizing,
-        }
-    }
+
+// Common struct used in Payment, Capture, Void, Refund
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct PowertranzBaseResponse {
+    transaction_type: u8,
+    approved: bool,
+    transaction_identifier: String,
+    original_trxn_identifier: Option<String>,
+    errors: Option<Vec<Error>>,
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct PowertranzPaymentsResponse {
-    status: PowertranzPaymentStatus,
-    id: String,
+impl ForeignFrom<(u8, bool)> for enums::AttemptStatus {
+    fn foreign_from((transaction_type, approved): (u8, bool)) -> Self {
+        match transaction_type {
+            // Auth
+            1 => match approved {
+                true => Self::Authorized,
+                false => Self::Failure,
+            },
+            // Sale or Capture
+            2 | 3 => match approved {
+                true => Self::Charged,
+                false => Self::Failure,
+            },
+            // Void
+            4 => match approved {
+                true => Self::Voided,
+                false => Self::VoidFailed,
+            },
+            // Refund
+            5 => match approved {
+                true => Self::AutoRefunded,
+                false => Self::Failure,
+            },
+            // Risk Management
+            _ => match approved {
```

Comment:
Will this be the case for any transaction type > 5?
Would be good to have a known upper cap for transaction type (say 10, this would depend on connector docs of course) beyond which we consider it an invalid transaction type.

Reply:
1,2,3,4,5 are the possible values

Reply:
Should we then consider raising an error if anything more than 5 is sent?

Reply:
I understand. What would happen if the connector does end up sending transaction type other than these values?

Won't it be better to error out as invalid transaction type instead of mapping status to pending/failure?